packages/core-modules/hardhat.config.js

```diff
@@ -3,7 +3,15 @@ require('@nomiclabs/hardhat-ethers');
 require('@synthetixio/deployer');
 
 module.exports = {
-  solidity: '0.8.4',
+  solidity: {
```

Comment:
Making solc compiler settings more explicit to avoid bytecode differences in the future.